http-generator-helidon/src/main/java/io/avaje/http/helidon/Filter.java

```diff
@@ -0,0 +1,14 @@
+package io.avaje.http.helidon;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Marks a method as a Helidon Filter
+ */
+@Target(METHOD)
+@Retention(SOURCE)
+public @interface Filter {}
```

Comment:
I'd expect the Before/After/Filter annotations to be in avaje-http-api.

Why is this not in avaje-http-api?

Reply:
It basically makes it clear that the `Before/After` functionality is Javalin only, and vice versa for Helidon Filter.

Reply:
But putting it into the generator is worse no?  We don't want the application controllers etc to have a dependency on the generators.

We might have a problem with Before/After being Javalin specific (at the moment) and Filter being Helidon specific but maybe we have options for dealing with that - like an explicit exception when defined but not supported etc.

These annotations can't be in the generators though right?

Reply:
For myself, I'd rather these annotations where in avaje-http-api ... and we dealt with the issue of being Javalin or Helidon specific in some other way (like the generators saying that we can't use those annotations with that particular target).

Reply:
We already say that the generators should be added as provided dependencies, with the annotations designated with `Retention.SOURCE` I don't see an issue, since there is still no hard dependency on the generators

Reply:
I suppose I'm not keen on having "useless" classes hanging around and potentially causing confusion.

Reply:
there is also the matter of javalin adding more lifecycle methods `beforeMatched`/`afterMatched` so the amount of "useless" annotations will increase.

Reply:
How about we move `@Filter` into avaje-http-api.  That will keep me happy wrt Helidon Nima and I think `@Filter` is _"generally correct / expected"_.

We can keep the Javalin annotations where they are for now, maybe move them into a separate avaje-javalin-api later.